modules/node-role-management.adoc

```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// * machine_management/managing-infrastructure-machinesets.adoc
+
+[id="node-role-management_{context}"]
+= Node role management
+
+By default, all nodes, excluding those on the control plane, are labeled with the `node-role.kubernetes.io/worker: ""` role. Nodes with just the `worker` role are called worker, or compute nodes. To create a specialized non-compute node that is not on the control plane, like an infrastructure or app node, it is recommended to keep the `worker` role and define the additional role too. For example, for an infra node, it should be labeled with the `node-role.kubernetes.io/worker: ""` and `node-role.kubernetes.io/infra: ""` roles. You can remove the `worker` role and only apply the specialized node role, but that requires creating a custom pool to allow for node upgrades due to constraints with the Machine Config Operator (MCO).
+
+You do not need a custom pool unless a specific node needs to be updated to have a different set of machine configs applied to it. You do not need a custom machine config pool or machine config for infrastructure nodes to work correctly, as it is not a strict requirement for tasks like node labeling, scheduling specific workloads, isolation, and preventing workload scheduling. You can, however, use a custom machine config pool if you find it useful for organizing and managing your machine configs. Visit the "Creating infrastructure machines with a dedicated configuration" section for more information.
```

Comment:
I favor removing this in exchange for the simpler statement "Optionally, if you would like to remove the worker label from your infrastructure nodes, you must create a MachineConfigPool specific to your infrastructure nodes."